make-rules/shared-macros.mk

```diff
@@ -345,6 +348,39 @@ export PARFAIT_NATIVESUNCXX=$(SPRO_VROOT)/bin/CC
 export PARFAIT_NATIVEGCC=$(GCC_ROOT)/bin/gcc
 export PARFAIT_NATIVEGXX=$(GCC_ROOT)/bin/g++
 
+# This should evaluate to empty string or a pathname like /usr/bin/ccache
+# depending on current PATH value and "which" implementation.
+# The assignment via ":=" is important, to only do this once in a Makefile,
+# and not on every reference to the value as "=" assignment would result in.
+# Review `man ccache` for optional configuration tuning, like cache size etc.
+# For production builds or suspected errors you can disable this feature by
+# setting CCACHE_DISABLE=true (as an environment or makefile variable value).
+# In fact, if you want to use this feature, you must set CCACHE_DISABLE=false
+# Still, absence of ccache in PATH is not considered a fatal error since the
+# build would just proceed well with original compiler.
+export CCACHE := $(shell \
+    if test -n "$(CCACHE)" ; then \
+        echo "$(CCACHE)"; \
+    else \
+        if test x"$${CCACHE_DISABLE-}" = xtrue -o x"$(CCACHE_DISABLE)" = xtrue ; then \
```

Comment:
As a ccache setting envvar which we abuse here (and duplicate as a
make-var), it has two valid values: false and true. And empty which is
equivalent to false (e.g. by default if you execute ccache - you want it to
act).

Per your suggestion, for us the default is true (disable and do not even
call ccache unless requested otherwise).

The current logic QUIETLY does nothing if the variable is
empty/unset/invalid, and VERBOSELY does or doesn't (try to) enable ccache
if the value is explicit and valid.

--
Typos courtesy of GMail on my Samsung Android
31.03.2016 17:46 пользователь "Alexander Pyhalov" <notifications@github.com>
написал:

> In make-rules/shared-macros.mk
> <https://github.com/OpenIndiana/oi-userland/pull/1806#discussion_r58077354>
> :
>
> > @@ -345,6 +348,39 @@ export PARFAIT_NATIVESUNCXX=$(SPRO_VROOT)/bin/CC
> >  export PARFAIT_NATIVEGCC=$(GCC_ROOT)/bin/gcc
> >  export PARFAIT_NATIVEGXX=$(GCC_ROOT)/bin/g++
> >
> > +# This should evaluate to empty string or a pathname like /usr/bin/ccache
> > +# depending on current PATH value and "which" implementation.
> > +# The assignment via ":=" is important, to only do this once in a Makefile,
> > +# and not on every reference to the value as "=" assignment would result in.
> > +# Review `man ccache` for optional configuration tuning, like cache size etc.
> > +# For production builds or suspected errors you can disable this feature by
> > +# setting CCACHE_DISABLE=true (as an environment or makefile variable value).
> > +# In fact, if you want to use this feature, you must set CCACHE_DISABLE=false
> > +# Still, absence of ccache in PATH is not considered a fatal error since the
> > +# build would just proceed well with original compiler.
> > +export CCACHE := $(shell \
> > +    if test -n "$(CCACHE)" ; then \
> > +        echo "$(CCACHE)"; \
> > +    else \
> > +        if test x"$${CCACHE_DISABLE-}" = xtrue -o x"$(CCACHE_DISABLE)" = xtrue ; then \
>
> true seems to be not necessary here, as it's disabled by default
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/1806/files/afd705946bde4534a485216ee7c873c043923862#r58077354>
>